pandas/tseries/tools.py

```diff
@@ -510,6 +510,9 @@ def coerce(values):
         # we allow coercion to if errors allows
         return to_numeric(values, errors=errors)
 
+    # prevent overflow in case of int8 or int16
+    arg = arg.astype('int64', copy=False)
```

Comment:
strings are already tested / work. 

internally ``pd.to_numeric`` is used which coerces non-numeric to numeric (or it raises if it fails). Thus you are left with float/int.